### PR TITLE
Added parameters closeShape and closePath to ExtrudeShape and ExtrudeShapeCustom

### PR DESCRIPTION
**Pull request by BabylonJSGuide**
Ref: https://forum.babylonjs.com/t/normal-discontinuity-in-extrusions/26091

### CODE REVIEW

File: src/Meshes/Builders/shapeBuilder.ts [hunk 1]
```diff
@@ -27,7 +29,7 @@ import { Path3D } from "../../Maths/math.path";
  */
 export function ExtrudeShape(
     name: string,
-    options: { shape: Vector3[]; path: Vector3[]; scale?: number; rotation?: number; cap?: number; updatable?: boolean; sideOrientation?: number; frontUVs?: Vector4; backUVs?: Vector4; instance?: Mesh; invertUV?: boolean },
+    options: { shape: Vector3[]; path: Vector3[]; scale?: number; rotation?: number; closeShape?: boolean; closePath?: boolean; cap?: number; updatable?: boolean; sideOrientation?: number; frontUVs?: Vector4; backUVs?: Vector4; instance?: Mesh; invertUV?: boolean },
```
Comment (sebavan) [inline src/Meshes/Builders/shapeBuilder.ts:32]: This is unfortunately a breaking change, could both params be added at the end of the function ?

File: src/Meshes/Builders/shapeBuilder.ts [hunk 2]
```diff
@@ -78,6 +84,8 @@ export function ExtrudeShapeCustom(
         rotationFunction?: Nullable<{ (i: number, distance: number): number }>;
         ribbonCloseArray?: boolean;
         ribbonClosePath?: boolean;
+        closeShape?: boolean;
```
Comment (sebavan) [inline src/Meshes/Builders/shapeBuilder.ts:87]: Same comment for this one ?

File: src/Meshes/Builders/shapeBuilder.ts [hunk 3]
```diff
@@ -100,8 +108,8 @@ export function ExtrudeShapeCustom(
         (() => {
             return 0;
         });
-    const ribbonCloseArray = options.ribbonCloseArray || false;
-    const ribbonClosePath = options.ribbonClosePath || false;
+    const ribbonCloseArray = options.closePath || options.ribbonCloseArray || false;
```
Comment (sebavan) [inline src/Meshes/Builders/shapeBuilder.ts:111]: Why adding an extra param if it does identical than the existing one ?

**Review by sebavan — CHANGES_REQUESTED** [review]

File: src/Meshes/Builders/shapeBuilder.ts (hunk 1 above)
Comment (BabylonJSGuide) [reply src/Meshes/Builders/shapeBuilder.ts:32]: This change does not break existing code they only come into action when specifically set, defaults match existing code. Order does not matter as they are properties of the options parameter object not a parameter of the function.

File: src/Meshes/Builders/shapeBuilder.ts (hunk 2 above)
Comment (BabylonJSGuide) [reply src/Meshes/Builders/shapeBuilder.ts:87]: As above

**Review by Popov72 — APPROVED** [review]

**Pull request merged**